components/Server.coffee

```diff
@@ -16,24 +16,30 @@ exports.getComponent = ->
       error:
         datatype: 'object'
 
+  c.autoOrdering = false
   c.servers = []
+  c.context = null
 
-  c.on 'end', ->
+  c.tearDown = (done) ->
     for server in c.servers
       server._connections = 0
       server.close()
+    c.context.deactivate()
+    c.emit 'end'
```

Comment:
Don't do this here. The shutdown method (which calls teardown) will do it for you:

https://github.com/noflo/noflo/blob/master/src/lib/Component.coffee#L103